backend/app/controllers/uoas.js

```diff
@@ -91,34 +88,95 @@ module.exports = {
         }
 
         co(function* () {
-            var added = yield thunkQuery(
-                'SELECT name, id FROM "UnitOfAnalysis" ' +
+            // Check if Subjects already exist in DB
```

Comment:
I would check before `if (req.body.subjects) {` to ensure that none of the subjects have a blank name. IE `[{name: ''}]`. It's theoretically possible for a user to submit multiple elements in the array ( `[{name: ''}, {name: 'Shangra-La']`), and anyone can be blank, so do a check against each of the array entries, and just remove it if you find it. If the array is empty after a removal, throw a "no blank subject" error back.

Reply:
I added a block to check if an empty string made it through before doing the query